crates/compiler/tests/parser_test/mod.rs

```diff
@@ -1,9 +1,137 @@
+use std::fs::read_to_string;
+
 use super::common::{serialize_yaml, SourceLocation, TestErrorHandler};
 use super::tokenizer_test::{base_scan, Attribute, AttributeValue};
-use compiler::parser::{self, AstRoot, Element, ParseOption, Parser};
+use compiler::parser::{self, ParseOption, Parser};
 use insta::assert_snapshot;
 use serde::Serialize;
 
+#[derive(Serialize)]
+pub struct AstRoot {
+    pub children: Vec<AstNode>,
+    pub location: SourceLocation,
+}
+
+impl<'a> From<parser::AstRoot<'a>> for AstRoot {
+    fn from(root: parser::AstRoot<'a>) -> Self {
+        Self {
+            children: root
+                .children
+                .into_iter()
+                .map(|child| child.into())
+                .collect(),
+            location: root.location.into(),
+        }
+    }
+}
+
+#[derive(PartialEq, Eq, Serialize)]
+pub enum ElementType {
+    Plain,
+    Component,
+    Template,
+    SlotOutlet,
+}
+
+impl From<parser::ElementType> for ElementType {
+    fn from(ty: parser::ElementType) -> Self {
+        ty.into()
```

Comment:
This is not correct since the default `ty.into` will find the `From::from` trait method to generate the result, which incurs the stackoverflow.
Pattern matching is the standard way to implement the equivalent feature.